src/actions/walletActions.js

```diff
@@ -261,16 +125,9 @@ export const encryptAndSaveWalletAction = (
   backupStatus: BackupStatus,
   enableBiometrics: boolean = false,
 ) => {
-  return async (dispatch: Dispatch, getState: GetState) => {
-    const { wallet: { walletState } } = getState();
-    const { isImported, isBackedUp, isRecoveryPending } = backupStatus;
+  return async (dispatch: Dispatch) => {
+    dispatch({ type: SET_WALLET_IS_ENCRYPTING, payload: true });
 
-    // might be already in ENCRYPTING state, i.e. on pin change
-    if (walletState !== ENCRYPTING) {
-      dispatch({ type: UPDATE_WALLET_STATE, payload: ENCRYPTING });
-    }
-
-    await delay(50);
```

Comment:
Really happy to be seeing these removed 👍

Reply:
🔥 🔥 🔥